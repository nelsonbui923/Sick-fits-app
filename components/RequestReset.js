import React, {Component} from 'react';
import {Mutation} from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage';

const REQUEST_RESET_MUTATION =gql`
    mutation REQUEST_RESET_MUTATION($email: String!) {
        requestReset(email: $email) {
            message
        }
    }
`;

export default class ReqestReset extends Component {
    state = {
        email: '',
    };

    saveToState = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }

    render() {
        return (
            <Mutation 
            mutation={REQUEST_RESET_MUTATION} 
            variables={this.state}>
                {(reset, {error, loading, called}) => {
                return (
                <Form method="post"
                data-test="form" 
                onSubmit={async e => {
                    e.preventDefault();
                    await reset();
                    this.setState({ email: ''})
                }}>
                <fieldset disabled={loading} aria-busy={loading}>
                    <h2>Change your password</h2>
                    <Error error={error} />
                    {!error && !loading && called && <p>Success! Chceck your email for link.</p>}
                    <label htmlFor="email">
                        Email
                        <input type="email" name="email" placeholder="Email" value={this.state.email} onChange={this.saveToState} />
                    </label>
                    <button type="Submit">Request a reset!</button>
                </fieldset>
            </Form>)
                }}
            </Mutation>
        )
    }
}

export {REQUEST_RESET_MUTATION};