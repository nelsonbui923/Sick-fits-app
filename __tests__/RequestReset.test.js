import { mount } from 'enzyme';
import wait from 'waait';
import RequestReset, {REQUEST_RESET_MUTATION} from '../components/RequestReset';
import { MockedProvider } from 'react-apollo/test-utils';
import toJSON from 'enzyme-to-json';


const mocks = [
    {
        request: {
            query: REQUEST_RESET_MUTATION,
            variables: { email: 'nelson.bui923@gmail.com' },
        },
        result: {
            data: { requestReset: { message: 'success', __typename: 'Message' } },
        },
    },
];

describe('<RequestReset/>', () => {
    it('renders and matches snapshot', async () => {
        const wrapper = mount(
            <MockedProvider>
                <RequestReset />
            </MockedProvider>
        );
        const form = wrapper.find('form[data-test="form"]');
        expect(toJSON(form)).toMatchSnapshot();
    });


    it('calls the mutation', async () => {
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <RequestReset />
            </MockedProvider>
        );
        // simulate typing an email
        wrapper
            .find('input')
            .simulate('change',
            { target : { name: 'email', value: 'nelson.bui923@gmail.com' } 
            });
        // submit form
        wrapper.find('form').simulate('submit');
        await wait();
        wrapper.update();
        expect(wrapper.find('p').text()).toContain('Success! Chceck your email for link.');
    })
})