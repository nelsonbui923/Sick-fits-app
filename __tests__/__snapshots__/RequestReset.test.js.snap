// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Change your password
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </label>
    <button
      type="Submit"
    >
      Request a reset!
    </button>
  </fieldset>
</form>
`;
